qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,964 @@
+        """
+        super().__init__(name=name, **kwargs)
+        self.g = gclib.py()
+        self.address = address
+        self.open()
+
+    def open(self) -> None:
+        """
+        Open connection to Galil motion controller. This method assumes that
+        the initial mapping of Galil motion controller's hardware's mapping
+        to an IP address is done using GDK and the IP address in burned in.
+        This applies that Motion controller no more requests for an IP address
+        and a connection to the Motion controller can be done by the IP
+        address burned in.
+        """
+        self.g.GOpen(self.address + " --direct -s ALL")
+
+    def get_idn(self) -> Dict[str, Optional[str]]:
+        """
+        Get Galil motion controller hardware information
+        """
+        data = self.g.GInfo().split(" ")
+        idparts: List[Optional[str]] = [
+            "Galil Motion Control, Inc.",
+            data[1],
+            data[4],
+            data[3][:-1],
+        ]
+
+        return dict(zip(("vendor", "model", "serial", "firmware"), idparts))
+
+    def write_raw(self, cmd: str) -> None:
+        """
+        Write for Galil motion controller
+        """
+        self.g.GCommand(cmd + "\r")
+
+    def ask_raw(self, cmd: str) -> str:
+        """
+        Asks/Reads data from Galil motion controller
+        """
+        return self.g.GCommand(cmd + "\r")
+
+    def timeout(self, val: int) -> None:
+        """
+        Sets timeout for the instrument
+
+        Args:
+            val: time in milliseconds. -1 disables the timeout
+        """
+        if val < -1:
+            raise RuntimeError("Timeout value cannot be set less than -1")
+
+        self.g.GTimeout(val)
+
+    def close(self) -> None:
+        """
+        Close connection to the instrument
+        """
+        self.g.GClose()
+
+    def motion_complete(self, axes: str) -> None:
+        """
+        Wait for motion to complete for given axes
+        """
+        self.g.GMotionComplete(axes)
+
+
+class VectorMode(InstrumentChannel):
+    """
+    Class to control motors in vector mode
+    """
+
+    def __init__(self,
+                 parent: "DMC4133Controller",
+                 name: str,
+                 **kwargs: Any) -> None:
+        """
+        Initializes the vector mode submodule for the controller
+
+        Args:
+            parent: an instance of DMC4133Controller
+            name: name of the vector mode plane
+        """
+        super().__init__(parent, name, **kwargs)
+        self._plane = name
+        self._vector_position_validator = Ints(min_value=-2147483648,
+                                               max_value=2147483647)
+
+        self.add_parameter(
+            "coordinate_system",
+            get_cmd="CA ?",
+            get_parser=self._parse_coordinate_system_active,
+            set_cmd="CA {}",
+            vals=Enum("S", "T"),
+            docstring="sets coordinate system for the motion",
```

Comment:
shall the docstring also explain the values of S and T?

Reply:
ok